api/src/main/java/org/motechproject/nms/api/web/BaseController.java

```diff
@@ -191,6 +192,20 @@ protected State getStateForFrontLineWorker(FrontLineWorker flw, Circle circle) {
         State state = frontLineWorkerService.getState(flw);
 
         if (state == null && circle != null) {
+            state = getStateFromCircle(circle);
+        }
+
+        return state;
+    }
+
+
+
+
+    protected State getStateFromCircle(Circle circle) {
+
+        State state = null;
+
+        if (circle != null) {
             List<State> states = circle.getStates();
 
             if (states.size() == 1) {
```

Comment:
Just to be clear, if there is more than one state do we consider it deployed if it is deployed in one, or not deployed?  The code seems to assume deployed and this thread doesn't seem to have a conclusive answer.